### PR TITLE
Mod updates

### PR DESCRIPTION
**Pull request by TheGoatGod**
#### Summary

SUMMARY: Mods "updates some names, migration and more"

#### Purpose of change

updates old code

#### Describe the solution

apply this patch

#### Describe alternatives you've considered

don't apply the patch

#### Testing

all mods post to character creation screen

### CODE REVIEW

**Comment by TheGoatGod** [comment]
still, don't get the pr validator

**Comment by TheGoatGod** [comment]
its not a complete update of names but its a few

**Comment by Fris0uman** [comment]
```
SUMMARY: Mods "updates some names, migration and more"
```

**Comment by TheGoatGod** [comment]
are there any mods that should be included in the base game from my pack? i can add them

File: data/mods/Magiclysm/items/enchanted_belts.json [hunk 1]
```diff
@@ -57,7 +57,6 @@
         "pocket_type": "CONTAINER",
         "max_contains_volume": "3 L",
         "max_contains_weight": "9 kg",
-        "max_item_length": "70 cm",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/items/enchanted_belts.json:60]: why did you delete this stuff?

**Comment by KorGgenT** [comment]
as far as mainlining mods, that's a decision for you to make rather than us. make sure it follows the guidelines in https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/IN_REPO_MODS.md and you'll be golden

**Comment by KorGgenT** [comment]
also don't do that in this same pr there's enough going on here in the first place.

File: data/mods/Magiclysm/items/enchanted_belts.json [hunk 2]
```diff
@@ -114,7 +113,6 @@
         "pocket_type": "CONTAINER",
         "max_contains_volume": "5 L",
         "max_contains_weight": "12 kg",
-        "max_item_length": "125 cm",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/items/enchanted_belts.json:117]: why did you delete this?

File: data/mods/Magiclysm/items/enchanted_belts.json [hunk 3]
```diff
@@ -184,14 +182,7 @@
...
     "pocket_data": [
-      {
-        "pocket_type": "CONTAINER",
-        "max_contains_volume": "1 L",
-        "max_contains_weight": "3 kg",
-        "max_item_length": "70 cm",
-        "moves": 2,
-        "holster": true
-      }
+      { "pocket_type": "CONTAINER", "max_contains_volume": "1 L", "max_contains_weight": "3 kg", "moves": 2, "holster": true }
```
Comment (KorGgenT) [inline data/mods/Magiclysm/items/enchanted_belts.json:185]: why did you remove the item length here?

File: data/mods/Magiclysm/items/enchanted_belts.json (hunk 1 above)
Comment (TheGoatGod) [reply data/mods/Magiclysm/items/enchanted_belts.json:60]: I thought I didn't merge this stuff let me undo

File: data/mods/Magiclysm/items/enchanted_belts.json (hunk 3 above)
Comment (TheGoatGod) [reply data/mods/Magiclysm/items/enchanted_belts.json:185]: didn't realize I did thank you for making me notice

File: data/mods/Magiclysm/items/enchanted_belts.json (hunk 2 above)
Comment (TheGoatGod) [reply data/mods/Magiclysm/items/enchanted_belts.json:117]: fixing

**Comment by TheGoatGod** [comment]
i will see what good enough talk with a few people and see what we come up with, regarding adding new mods

**Comment by Fris0uman** [comment]
> i think adding in all the "no" mods would be a good idea but one thing at a time

If by that you mean mod disabling content, like "No acid zombies mod" that's exactly the kind of things that doesn't fit https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/IN_REPO_MODS.md

File: data/mods/Aftershock/items/items.json [hunk 4]
```diff
@@ -17,7 +17,8 @@
   {
     "id": "afs_basin",
     "type": "GENERIC",
-    "name": { "ctxt": "container", "str": "basin" },
+    "name": { "str": "basin" },
+    "category": "container",
```
Comment (Qrox) [inline data/mods/Aftershock/items/items.json:21]:
```suggestion
    "name": { "ctxt": "container", "str": "basin" },
```
"ctxt" means the string is translated with a context to distinguish it from other strings with the same text but different meaning.

File: data/mods/Aftershock/items/comestibles/comestibles.json [hunk 5]
```diff
@@ -2,7 +2,7 @@
   {
     "id": "afs_sungel",
     "type": "COMESTIBLE",
-    "name": "sungel",
+    "name": { "str": "sungel" },
```
Comment (Qrox) [inline data/mods/Aftershock/items/comestibles/comestibles.json:5]:
```suggestion
    "name": { "str_sp": "sungel" },
```
For identical singular and plural forms

File: data/mods/Aftershock/items/comestibles/comestibles.json [hunk 6]
```diff
@@ -22,7 +22,7 @@
   {
     "id": "afs_sunesthesia",
     "type": "COMESTIBLE",
-    "name": "sunesthesia",
+    "name": { "str": "sunesthesia" },
```
Comment (Qrox) [inline data/mods/Aftershock/items/comestibles/comestibles.json:25]:
```suggestion
    "name": { "str_sp": "sunesthesia" },
```

File: data/mods/CRT_EXPANSION/items/crt_ammo.json [hunk 7]
```diff
@@ -3,7 +3,7 @@
     "type": "AMMO",
     "id": "pellet",
     "price": 1000,
-    "name": "lead pellets",
+    "name": { "str": "lead pellets" },
```
Comment (Qrox) [inline data/mods/CRT_EXPANSION/items/crt_ammo.json:6]:
```suggestion
    "name": { "str_sp": "lead pellets" },
```
And the same for other pellets.

File: data/mods/Dark-Skies-Above/overrides/items/carnivore.json [hunk 8]
```diff
@@ -18,7 +18,7 @@
   {
     "type": "COMESTIBLE",
     "id": "dry_meat_tainted",
-    "name": "dehydrated alien meat",
+    "name": { "str": "dehydrated alien meat" },
```
Comment (Qrox) [inline data/mods/Dark-Skies-Above/overrides/items/carnivore.json:21]:
```suggestion
    "name": { "str_sp": "dehydrated alien meat" },
```
The same for other meats/fats/tallows/lards

File: data/mods/CRT_EXPANSION/deadspace/deadspaceitems.json [hunk 9]
```diff
@@ -3,8 +3,8 @@
     "id": "ds_monitor",
     "type": "TOOL_ARMOR",
     "category": "armor",
-    "name": "CRIT S-I G.E.A.R",
-    "description": "C.R.I.T standard issue General Engineering Assistance Rig.  Plugged into your spinal cord, this device improves your overall physique and provides basic information on your surroundings.",
+    "name": { "str": "CRIT S-I G.E.A.R" },
+    "description": "C.R.I.T standard issue General Engineering Assistance Rig. Plugged into your spinal cord, this device improves your overall physique and provides basic information on your surroundings.",
```
Comment (ZhilkinSerg) [inline data/mods/CRT_EXPANSION/deadspace/deadspaceitems.json:7]:
```suggestion
    "description": "C.R.I.T standard issue General Engineering Assistance Rig.  Plugged into your spinal cord, this device improves your overall physique and provides basic information on your surroundings.",
```

File: data/mods/CRT_EXPANSION/items/crt_ammo.json [hunk 10]
```diff
@@ -23,7 +23,7 @@
     "id": "dhp_pellet",
     "copy-from": "pellet",
     "type": "AMMO",
-    "name": "domed HP pellets",
+    "name": { "str": "domed HP pellets" },
```
Comment (Qrox) [inline data/mods/CRT_EXPANSION/items/crt_ammo.json:26]: You only applied the suggestion in my previous comment but did not update the other instances. All these "xxx_pellets" should use "str_sp" instead of "sp".

File: data/mods/Dark-Skies-Above/overrides/items/carnivore.json [hunk 11]
```diff
@@ -27,22 +27,22 @@
...
+    "name": { "str": "alien bone" },
     "description": "A piece of a hard material with many crests and protrusions, made of a hard material that occasionally leaves a bit of moisture on your hand when you touch it.  It seems basically like a bone, though not one you'd see in any anatomy textbook.  You could eat it, but no doubt it would not only taste henious but also make you sick, being made of material your body was never supposed to injest.  It might at least be able to be used for something, if not as well as the earthly variant due to its fiddly size and shape."
   },
   {
     "type": "COMESTIBLE",
     "id": "fat_tainted",
     "category": "other",
     "copy-from": "fat_tainted",
-    "name": "alien fat",
+    "name": { "str": "alien fat" },
```
Comment (Qrox) [inline data/mods/Dark-Skies-Above/overrides/items/carnivore.json:38]: And all these "xxx fat", "xxx tallow", "xxx meat", "xxx lard" should use "str_sp" too.

File: data/mods/CRT_EXPANSION/items/crt_ammo.json (hunk 10 above)
Comment (TheGoatGod) [reply data/mods/CRT_EXPANSION/items/crt_ammo.json:26]: "str_sp" instead of "sp".? I never did them all these mods are maintained and the mod author should be updating these, I'm only adding my additions, I have too many mods to update.

**Comment by TheGoatGod** [comment]
leaving this as is, moving on to mods not in the repo if you don't merge that's fine

**Comment by anothersimulacrum** [comment]
If you're abandoning this PR, I'm just going to mark it as stale and close it.

File: data/mods/CRT_EXPANSION/items/crt_ammo.json (hunk 10 above)
Comment (Qrox) [reply data/mods/CRT_EXPANSION/items/crt_ammo.json:26]: "str_sp" instead of "str" I mean, sorry. You're changing names from strings to objects already so I think it would be better if you changed it to the most appropriate ones in this PR.
